### DISCUSSION TITLE
Passing a resource via RunRequest.run_config

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [dagster-io/dagster#18597] [Community]
Hi, I have a sensor that detects files in an s3 bucket, I want to process this file in a software-defined asset using the same S3Resource that the sensor uses. I have tried passing the resource as a property of a run config and I get a weird error

```
class RawFilesConfig(Config):
    s3_resource: S3Resource
    s3_bucket: str
```

```
    return SensorResult(
            dynamic_partitions_requests=[raw_files.build_add_request(new_s3_keys)],
            run_requests=[
                RunRequest(
                    partition_key=key,
                    run_config=RunConfig(
                        {
                            "raw_file": RawFilesConfig(
                                s3_resource=s3,
                                s3_bucket=s3_bucket,
                                )
                            }
                        )
                    )
                for key in new_s3_keys
                ],
            cursor=last_key,
            )
```

```
dagster._core.errors.DagsterInvalidConfigError: Error in config for job raw_file_job
    Error 1: Received unexpected config entry "s3_resource" at path root:ops:raw_file:config. Expected: "{ s3_bucket: (String | { env: String }) }".
```

If I change the `s3_resource` property to be a string and set it to the name of the resource rather than the object, it works. Is it possible to pass the S3Resource as an object? If not how can I ensure that the asset uses the same S3Resource as the sensor?

**Comment by niteesh-cashflo** [comment]
were you able to fix this?

**Reply by the4thamigo-uk** [reply]
No, I decided just to make sure the asset was configured with the same resource. However, I now know that you can pass resources in the [run_request](https://docs.dagster.io/_apidocs/config#dagster.RunConfig), so if I wanted to do this Id try it that way.
